File: web/src/layout/ComponentRowQuestionXml.ts
//
// ComponentRowQuestionXml.ts
// Created by Khrysalis XML Typescript
//
import { loadHtmlFromString, findViewById, getViewById, replaceViewWithId } from 'khrysalis/dist/views/html'
import { customViewSetDelegate } from 'khrysalis/dist/views/CustomView.actual'
import htmlForDefault from './component_row_question.html'
//! Declares com.lightningkite.khrysaliswebsite.layouts.ComponentRowQuestionXml
export class ComponentRowQuestionXml {
    xmlRoot!: HTMLElement;
    container!: HTMLDivElement;
    question!: HTMLElement;
    loadHtmlString(): string {
        return htmlForDefault;
    }
    setup(dependency: Window): HTMLElement {
        const view = loadHtmlFromString(this.loadHtmlString());
        this.xmlRoot = view
        this.container = getViewById<HTMLDivElement>(view, "container");
        this.question = getViewById<HTMLElement>(view, "question");
        return view
    }
}

File: android/src/main/java/com/lightningkite/khrysaliswebsite/layouts/ComponentRowQuestionXml.kt
//
// ComponentRowQuestionXml.kt
// Created by Khrysalis XML Android
//
package com.lightningkite.khrysaliswebsite.layouts

import android.widget.*
import android.view.*
import com.lightningkite.khrysalis.views.*
import com.lightningkite.khrysaliswebsite.R

class ComponentRowQuestionXml {

    lateinit var container: LinearLayout
    lateinit var question: TextView
    
    
    lateinit var xmlRoot: View

    fun setup(dependency: ViewDependency): View {
        val view = LayoutInflater.from(dependency.context).inflate(R.layout.component_row_question, null, false)
        return setup(view)
    }
    fun setup(view: View): View {
        xmlRoot = view
        container = view.findViewById<LinearLayout>(R.id.container)
        question = view.findViewById<TextView>(R.id.question)
        
        
        
        return view
    }
}